### ISSUE TITLE
Running many tests runs into memory limit quickly

### ISSUE THREAD

**Issue by markuspf** [gap-packages/meataxe64#12]
Consider this code

```
gap> extract_roundtrip := function(n, q)
>    local m, mm;
>    m := RandomMat(n, n, GF(q));
>    mm := MTX64_ExtractMatrix(MTX64_Matrix(m));
>    if m <> mm then
>        Error("Mismatch between matrix and roundtrip matrix");
>    fi;
>    return true;
>    end;;
gap> for i in [1..100] do extract_roundtrip(Random([2..100]), Random(Primes) ^ Random([1..3])); od;;
```

This runs into a break loop because it hits the memory limit. It looks like `MemoizePosIntFunction` is at fault, because it uses a long list of unbound entries.

**Comment by markuspf** [comment]
This is due to the finite field cache using `MemoizePosIntFunction`. @ChrisJefferson suggests to use datastructures + hash tables for this.

**Comment by stevelinton** [comment]
@ChrisJefferson  is clearly correct. Memoize... was a quick initial hack.

**Comment by stevelinton** [comment]
Closed by #16
